edsnlp/pipelines/core/matcher/matcher.py

```diff
@@ -39,17 +40,28 @@ def __init__(
         regex: Optional[Patterns],
         attr: str,
         ignore_excluded: bool,
+        algorithm: str = "exact",
```

Comment:
Should we use Enum here ?

Reply:
I'm not used to enums, does pydantic validate them ? Maybe the enum should go in the factory definition but I've added validation in the __init__ methods anyway.